### PR TITLE
Implement new feature: initialise S3 Bento repository.

### PR DESCRIPTION
**Pull request by leonsim**
(Thanks for sending a pull request! Please make sure to read the contribution guidelines, then fill out the blanks below.)

What changes were proposed in this pull request?
------------------------------------------------
Implement new feature: initialise S3 Bento repository.

Does this close any currently open issues?
------------------------------------------
Nope

How was this patch tested?
--------------------------
Not yet.

### CODE REVIEW

File: bentoml/artifact/fastai_model_artifact.py [hunk 1]
```diff
@@ -46,8 +46,10 @@ def __init__(self, name):
             raise SystemError("fast ai requires python 3.6 version or higher")
 
         if fastai is None:
-            raise ImportError("fastai package is required to use "
-                              "bentoml.artifacts.FastaiModelArtifact")
+            raise ImportError(
+                "fastai package is required to use "
+                "bentoml.artifacts.FastaiModelArtifact"
+            )
```
Comment (leonsim) [inline bentoml/artifact/fastai_model_artifact.py]: Strange I didn't change this. Maybe my IDE auto changed this?

Comment (parano) [reply bentoml/artifact/fastai_model_artifact.py]: did you run the ./script/format.sh script?

File: bentoml/repository/__init__.py [hunk 2]
```diff
@@ -118,6 +122,21 @@ def __init__(self, base_url):
...
+        parse_result = urlparse(base_url)
+        bucket = parse_result.netloc
+        base_path = parse_result.path
+
+        s3_client = boto3.client("s3")
+
+        try:
+            filename = uuid.uuid4().hex
+            s3_path = os.path.join(base_path, filename)
+            s3_client.upload_file(Filename=filename, Bucket=bucket, Key=s3_path)
```
Comment (parano) [inline bentoml/repository/__init__.py]: Is this for testing users' write-permission? I think it's ok to just be optimistic here and only raise when it actually run into problems uploading things to s3

File: bentoml/artifact/fastai_model_artifact.py (hunk 1 above)
Comment (leonsim) [reply bentoml/artifact/fastai_model_artifact.py]: oh yeah, it is that script.

File: bentoml/repository/__init__.py (hunk 2 above)
Comment (leonsim) [reply bentoml/repository/__init__.py]: yes, ok, will remove this.

**Pull request merged**